File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.go
func twoEditWords(queries []string, dictionary []string) []string {
    ans := []string{}
    for _, query := range queries {
        for _, dict := range dictionary {
            if query == dict {
                ans = append(ans, query)
                break
            }

            count := 0
            for i := 0; i < len(query); i++ {
                if query[i] != dict[i] {
                    count++
                }
            }

            if count <= 2 {
                ans = append(ans, query)
                break
            }
        }
    }

    return ans
}

// array, brute force, string
// time: O(qdl)
// space: O(q)

File: 2452-words-within-two-edits-of-dictionary/2452-words-within-two-edits-of-dictionary.py
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        ans = []
        for q in queries:
            for d in dictionary:
                if q == d:
                    ans.append(q)
                    break
                
                edit = 0
                valid = True
                for i in range(len(q)):
                    if q[i] != d[i]:
                        edit += 1
                    if edit > 2:
                        valid = False
                        break
                
                if valid and edit <= 2:
                    ans.append(q)
                    break   
        
        return ans

# array, brute force, string
# time: O(qdl)
# space: O(q)